wolvic/wolvic_content_client.cc

```diff
@@ -0,0 +1,50 @@
+#include "wolvic/wolvic_content_client.h"
+
+#include "base/strings/string_piece.h"
+#include "base/strings/utf_string_conversions.h"
+#include "base/values.h"
+#include "build/build_config.h"
+#include "content/shell/grit/shell_resources.h"
+#include "ui/base/l10n/l10n_util.h"
+#include "ui/base/resource/resource_bundle.h"
+
+namespace content {
+
+WolvicContentClient::WolvicContentClient() {}
+
+WolvicContentClient::~WolvicContentClient() {}
+
+std::u16string WolvicContentClient::GetLocalizedString(int message_id) {
+  return l10n_util::GetStringUTF16(message_id);
+}
+
+base::StringPiece WolvicContentClient::GetDataResource(
+    int resource_id,
+    ui::ResourceScaleFactor scale_factor) {
+  return ui::ResourceBundle::GetSharedInstance().GetRawDataResourceForScale(
+      resource_id, scale_factor);
+}
+
+base::RefCountedMemory* WolvicContentClient::GetDataResourceBytes(
+    int resource_id) {
+  return ui::ResourceBundle::GetSharedInstance().LoadDataResourceBytes(
+      resource_id);
+}
+
+std::string WolvicContentClient::GetDataResourceString(int resource_id) {
+  return ui::ResourceBundle::GetSharedInstance().LoadDataResourceString(
+      resource_id);
+}
+
+gfx::Image& WolvicContentClient::GetNativeImageNamed(int resource_id) {
+  return ui::ResourceBundle::GetSharedInstance().GetNativeImageNamed(
+      resource_id);
+}
+
+void WolvicContentClient::AddAdditionalSchemes(Schemes* schemes) {
+#if BUILDFLAG(IS_ANDROID)
```

Comment:
I wonder if we need to keep IS_ANDROID checks, considering that we only support Android and we don't have any other platform-specific code.

Reply:
I don't have a strong opinion. @svillar, wdyt?

Reply:
In the context of Chromium I do think they make sense, but I guess the plan is to move this to Wolvic. Once inside Wolvic they don't make any sense indeed as Wolvic is thightly coupled to Android.

Reply:
I agree. We would need to override UserAgent as Wolvic's one through the Java interface. Nonetheless, I think it would be meaningful to have this code if we want to quickly check the issue of the UA.

Reply:
I think that we don't need IS_ANDROID so long as wolvic/BUILD.gn has assert(is_android).

Reply:
Removed IS_ANDROID